src/OrchardCore.Modules/OrchardCore.Setup/GraphQL/CreateTenantMutation.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GraphQL;
+using GraphQL.Resolvers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Apis.GraphQL.Arguments;
+using OrchardCore.Apis.GraphQL.Types;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Setup.Services;
+using OrchardCore.Setup.ViewModels;
+
+namespace OrchardCore.Setup.GraphQL
+{
+    public class CreateTenantMutation : MutationFieldType
+    {
+        private const string DefaultRecipe = "Default";
+        private readonly IEnumerable<DatabaseProvider> _databaseProviders;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CreateTenantMutation(
+            ShellSettings shellSettings,
+            IEnumerable<DatabaseProvider> databaseProviders,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            _databaseProviders = databaseProviders;
+
+            _httpContextAccessor = httpContextAccessor;
+            Name = "CreateTenant";
+
+            Arguments = new AutoRegisteringQueryArguments<SetupViewModel>();
+
+            Type = typeof(CreateTenantOutcomeType);
+
+            Resolver = new AsyncFieldResolver<object, object>(async (context) =>
+            {
+                var serviceProvider = httpContextAccessor.HttpContext.RequestServices;
```

Comment:
maybe better to use the field `_httpContextAccessor` initialized above, or just not declare it.